custom_model_runner/datarobot_drum/drum/gpu_predictors/base.py

```diff
@@ -139,7 +147,11 @@ def _chat(self, completion_create_params, association_id):
         if not model_name_from_request or model_name_from_request == self.DEFAULT_MODEL_NAME:
             completion_create_params["model"] = self.served_model_name
 
-        return self.ai_client.chat.completions.create(**completion_create_params)
+        # Separate valid kwargs from extra kwargs
+        valid_params = set(COMPLETIONS_CREATE_SIGNATURE.parameters.keys())
```

Comment:
`COMPLETIONS_CREATE_SIGNATURE` is defined only when _HAS_OPENAI is "True". Does this method need a check for this flag?

Reply:
So it's not the most pretty thing but in the object constructor, we have this check so I feel like it is ok in any method to then assume that the library is installed.